src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -101,6 +101,27 @@ public override bool Execute()
             return true;
         }
 
+        /// <remarks>
+        /// Filenames with special characters like '#' and'{' get written
+        /// into the section names in the resulting .editorconfig file. Later,
+        /// when the file is parsed in configuration options these special
+        /// characters are interpretted as invalid values and ignored by the
+        /// processor. We encode the special characters in these strings
+        /// before writing them here.
+        /// </remarks>
+
+        private static void EncodeString(StringBuilder builder, string value)
+        {
+            foreach (var c in value)
+            {
+                if (c is '*' or '?' or '{' or ',' or ';' or '}' or '[' or ']' or '#')
```

Comment:
>c is [](http://example.com/codeflow?start=20&length=4)

Do we need to include `!`?

Reply:
Yep, it is a wildcard character. I'll go ahead and add it.

Reply:
@cston @captainsafia What is the special meaning of `!`? It's treated currently as a simple character:

https://github.com/dotnet/roslyn/blob/481529aa563e63b059b50f826cd99f523d680f57/src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.SectionNameMatching.cs#L443